src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -941,10 +943,488 @@ class spell_silverpine_forsaken_trooper_masterscript_high_command : public Spell
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    {
+                        if (player->GetQuestStatus(QUEST_ITERATING_UPON_SUCCESS) == QUEST_STATUS_INCOMPLETE)
+                            me->GetMotionMaster()->MovePath(PATH_BAT_AROUND_LAKE, false);
+                    }
+                }
+                break;
+            }
+
+            case PATH_BAT_TO_HOME:
+            {
+                if (waypointId == WAYPOINT_LAST_POINT_TO_HOME)
+                {
+                    me->GetVehicleKit()->RemoveAllPassengers();
+
+                    me->DespawnOrUnsummon();
+                }
+                break;
+            }
+
+            default:
+                break;
+        }
+    }
+
+    void DoAction(int32 param) override
+    {
+        switch (param)
+        {
+            case ACTION_GO_HOME:
+            {
+                if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                {
+                    me->PauseMovement();
+
+                    me->GetMotionMaster()->Clear();
+
+                    me->SetSpeed(UnitMoveType::MOVE_RUN, 35.78791f);
+
+                    me->GetMotionMaster()->MovePath(PATH_BAT_TO_HOME, false);
+
+                    SetupFinishActionBar();
+
+                    player->VehicleSpellInitialize();
+
+                    Talk(TALK_BAT_GOING_HOME, player);
+                }
+                break;
+            }
+
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CHECK_FINISH_ITERATING:
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    {
+                        if (player->GetReqKillOrCastCurrentCount(QUEST_ITERATING_UPON_SUCCESS, NPC_VILE_FIN_ORACLE) == DATA_ITERATING_UPON_SUCCESS_QUEST_REQ)
+                            player->CastSpell(me, SPELL_GO_HOME, true);
+                        else
+                            _events.ScheduleEvent(EVENT_CHECK_FINISH_ITERATING, 500ms);
+                    }
+                    break;
+                }
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    void SetInitialActionBar()
+    {
+        me->m_spells[0] = SPELL_BLIGHT_CONCOCTION;
+        me->m_spells[1] = 0;
+    }
+
+    void SetCircularActionBar()
+    {
+        me->m_spells[0] = SPELL_BLIGHT_CONCOCTION;
+        me->m_spells[1] = SPELL_GO_HOME;
+    }
+
+    void SetupFinishActionBar()
+    {
+        me->m_spells[0] = 0;
+        me->m_spells[1] = 0;
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+};
+
+// Go Home - 83594
+class spell_silverpine_go_home : public SpellScript
```

Comment:
You can just use SpellHit in AI, easier + on retail dummies generally aren't scripted but handled in AI's SpellHit

Reply:
Sure, I'll handle it that way on the next update.

Reply:
Feedback applied!